bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/discovery/CaddxBridgeDiscovery.java

```diff
@@ -0,0 +1,174 @@
+@NonNullByDefault
+public class CaddxBridgeDiscovery extends AbstractDiscoveryService implements CaddxPanelListener {
+    private final Logger logger = LoggerFactory.getLogger(CaddxBridgeDiscovery.class);
+
+    private @NonNullByDefault({}) SerialPortManager portManager;
+    static final int[] BAUDRATES = { 9600, 19200, 38400, 57600, 115200 };
+    private volatile boolean bridgeFound = false;
+
+    /**
+     * Constructor.
+     */
+    public CaddxBridgeDiscovery() {
+        super(CaddxBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS, 15, true);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.trace("Start Caddx Bridge discovery.");
+
+        if (portManager != null) {
+            discoverBridge();
+        }
+    }
+
+    public synchronized void discoverBridge() {
+        logger.trace("Starting Caddx Bridge Discovery.");
+
+        Stream<SerialPortIdentifier> ports = portManager.getIdentifiers();
+        Iterator<SerialPortIdentifier> iterator = ports.iterator();
+        while (iterator.hasNext()) {
+            SerialPortIdentifier portIdentifier = iterator.next();
+
+            try {
+                for (int baudrate : BAUDRATES) {
+                    if (checkforBridge(CaddxProtocol.Binary, portIdentifier.getName(), baudrate)) {
+                        break;
+                    }
+                    if (checkforBridge(CaddxProtocol.Ascii, portIdentifier.getName(), baudrate)) {
+                        break;
+                    }
+                }
+            } catch (UnsupportedCommOperationException | PortInUseException | IOException
+                    | TooManyListenersException e1) {
+                logger.debug("Port: {} is not applicable.", portIdentifier.getName());
+                continue;
+            }
+
+            // if the thread was interrupted then exit the discovery loop
+            if (Thread.currentThread().isInterrupted()) {
+                break;
+            }
+        }
+    }
+
+    private boolean checkforBridge(CaddxProtocol protocol, String serialPort, int baudrate)
+            throws UnsupportedCommOperationException, PortInUseException, IOException, TooManyListenersException {
+        logger.debug("Checking protocol: {}, port: {}, baud: {}", protocol, serialPort, baudrate);
+
+        bridgeFound = false;
+        CaddxCommunicator caddxCommunicator = new CaddxCommunicator(portManager, protocol, serialPort, baudrate);
+        caddxCommunicator.addListener(this);
+        caddxCommunicator.transmit(new CaddxMessage(CaddxMessageType.Interface_Configuration_Request, ""));
+
+        // Wait for 4 seconds for a response
+        try {
+            TimeUnit.SECONDS.sleep(4);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+
+        caddxCommunicator.stop();
+        caddxCommunicator = null;
+        return bridgeFound;
```

Comment:
This always returns false?

Reply:
No. [It would not discover anything]
The discovery works by sending a request to the serial port and waiting for 4 seconds to receive a valid message. When a reply is received bridgeFound is set to true.

Reply:
bridgeFound is set to true at line 130.
Should i resolve this conversation?